File: generated-sources/kotlin-spring/mojang-sessions/src/main/kotlin/org/openapitools/model/PlayerProfileTexturePropertyValue.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.PlayerTexture
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * This is obtained by decoding the \"textures\" property
 * @param timestamp UNIX timestamp in milliseconds
 * @param profileId The player UUID without hyphens
 * @param profileName The player's name
 * @param signatureRequired Only present if unsigned was set to false in the request
 * @param textures 
 */
data class PlayerProfileTexturePropertyValue (

        @get:NotNull 
        @JsonProperty("timestamp") val timestamp: Long,

        @get:NotNull 
        @JsonProperty("profileId") val profileId: String,

        @get:NotNull 
        @JsonProperty("profileName") val profileName: String,

        @JsonProperty("signatureRequired") val signatureRequired: Boolean? = null,

        @JsonProperty("textures") val textures: PlayerTexture? = null
) {

}

File: generated-sources/javascript-closure-angular/mojang-sessions/API/Client/PlayerProfileTexturePropertyValue.js
goog.provide('API.Client.PlayerProfileTexturePropertyValue');

/**
 * This is obtained by decoding the \"textures\" property
 * @record
 */
API.Client.PlayerProfileTexturePropertyValue = function() {}

/**
 * UNIX timestamp in milliseconds
 * @type {!number}
 * @export
 */
API.Client.PlayerProfileTexturePropertyValue.prototype.timestamp;

/**
 * The player UUID without hyphens
 * @type {!string}
 * @export
 */
API.Client.PlayerProfileTexturePropertyValue.prototype.profileId;

/**
 * The player's name
 * @type {!string}
 * @export
 */
API.Client.PlayerProfileTexturePropertyValue.prototype.profileName;

/**
 * Only present if unsigned was set to false in the request
 * @type {!boolean}
 * @export
 */
API.Client.PlayerProfileTexturePropertyValue.prototype.signatureRequired;

/**
 * @type {!API.Client.PlayerTexture}
 * @export
 */
API.Client.PlayerProfileTexturePropertyValue.prototype.textures;

